src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/removeusercaseaccess/RemoveUserCaseAccessAboutToStartHandler.java

```diff
@@ -0,0 +1,96 @@
+
+    // Separator element for DynamicListElement code
+    static final String CODE_VALUES_SEPARATOR = "~#~";
+
+    private final AuthTokenGenerator authTokenGenerator;
+    private final DataStoreClient dataStoreClient;
+    private final IdamAuthApi idamAuthApi;
+
+    public RemoveUserCaseAccessAboutToStartHandler(FinremCaseDetailsMapper finremCaseDetailsMapper,
+                                                   AuthTokenGenerator authTokenGenerator,
+                                                   DataStoreClient dataStoreClient, IdamAuthApi idamAuthApi) {
+        super(finremCaseDetailsMapper);
+        this.authTokenGenerator = authTokenGenerator;
+        this.dataStoreClient = dataStoreClient;
+        this.idamAuthApi = idamAuthApi;
+    }
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, CaseType caseType, EventType eventType) {
+        return CallbackType.ABOUT_TO_START.equals(callbackType)
+            && EventType.REMOVE_USER_CASE_ACCESS.equals(eventType)
+            && (CaseType.CONSENTED.equals(caseType) || CaseType.CONTESTED.equals(caseType));
+    }
+
+    @Override
+    public GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> handle(
+        FinremCallbackRequest finremCallbackRequest, String authToken) {
+
+        String caseId = String.valueOf(finremCallbackRequest.getCaseDetails().getId());
+        log.info("Remove User Case Access about to start callback for Case ID: {}", caseId);
+
+        List<CaseAssignedUserRole> caseAssignedUserRoles = getCaseAssignedUserRoles(authToken, caseId);
+        List<DynamicListElement> dynamicListElements = caseAssignedUserRoles.stream()
+            .map(c -> getUserCaseAccess(authToken, c))
+            .map(this::map)
+            .toList();
+        DynamicList caseUserAccessList = DynamicList.builder()
+            .listItems(dynamicListElements)
+            .build();
+
+        FinremCaseData caseData = finremCallbackRequest.getCaseDetails().getData();
+        caseData.setUserCaseAccessList(caseUserAccessList);
+
+        return GenericAboutToStartOrSubmitCallbackResponse.<FinremCaseData>builder()
+            .data(caseData)
+            .build();
+    }
+
+    private List<CaseAssignedUserRole> getCaseAssignedUserRoles(String authToken, String caseId) {
+        String serviceToken = authTokenGenerator.generate();
+        CaseAssignedUserRolesResource caseAssignedUserRolesResource = dataStoreClient.getUserRoles(authToken,
+            serviceToken, caseId, null);
+
+        return caseAssignedUserRolesResource.getCaseAssignedUserRoles();
+    }
+
+    private UserCaseAccess getUserCaseAccess(String authToken, CaseAssignedUserRole caseAssignedUserRole) {
+        UserDetails userDetails = idamAuthApi.getUserByUserId(authToken, caseAssignedUserRole.getUserId());
+        return UserCaseAccess.of(caseAssignedUserRole, userDetails);
+    }
+
+    private DynamicListElement map(UserCaseAccess userCaseAccess) {
```

Comment:
Maybe the name should be more specific, something like `mapToDynamicListElement`.

Reply:
Method renamed